exchange/bidder.go

```diff
@@ -522,11 +523,27 @@ func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.Reques
 
 func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.RequestData, logger util.LogMsg, bidderRequestStartTime time.Time, tmaxAdjustments *TmaxAdjustmentsPreprocessed) *httpCallInfo {
 	var requestBody []byte
+	var gzipWriterPool = sync.Pool{
+		New: func() interface{} {
+			return gzip.NewWriter(nil)
+		},
+	}
 
 	switch strings.ToUpper(bidder.config.EndpointCompression) {
 	case Gzip:
-		requestBody = compressToGZIP(req.Body)
+		// Compress to GZIP
+		var b bytes.Buffer
+		w := gzipWriterPool.Get().(*gzip.Writer) // Utilize Sync Pool
+		w.Reset(&b)
+		w.Write([]byte(requestBody))
+		w.Close()
+		requestBody = b.Bytes()
```

Comment:
Is the `gzip.Writer` itself an io.Reader? Could we pass it directly into `http.NewRequesr` instead of getting the byte array and wrapping that in a byte buffer.

Reply:
After looking into it, it appears that `gzip.Writer` is not an io.Reader, so I think we have to utilize it as it is currently in order to pass the compressed GZIP request to the `http.NewRequest` function.